### ISSUE TITLE
Leading null characters in jsonlog in Stackdriver seen using jsonlog.so

### ISSUE THREAD

**Issue by naveen27232** [michaelpq/pg_plugins#27]
Hi Michael

I have tried using the “jsonlog.c” file shared at https://github.com/michaelpq/pg_plugins/tree/master/jsonlog 
and when I compile and use the jsonlog.so library in my postgres image , the Stackdriver logs show some extra 9 characters as null (?????????) as shown in the attached image  "null-char-Image.png" and "null__logs__2019-07-30T04-04.json".
Also attached-
jsonlog.c
jsonlog.so
postgresql.conf

Regards
Naveen Sehgal
Avaya
 
[Jsonlog.zip](https://github.com/michaelpq/pg_plugins/files/3485068/Jsonlog.zip)

**Comment by exe-dealer** [comment]
I have similar bug. Some messages are prefixed by chunk of previous message

```
{"timestamp":"2019-11-13T11:10:50.694Z","pid":1,"session_id":"5dcbe4ba.1","error_severity":"LOG","message":"starting PostgreSQL 12.0 on x86_64-pc-linux-gnu, compiled by gcc (Alpine 8.3.0) 8.3.0, 64-bit"}
{"timestamp":"2019-11-13T11:10:50.694Z","pid":1,"session_id":"5dcbe4ba.1","error_severity":"LOG","message":"listening on IPv4 address \"0.0.0.0\", port 5432"}
piled by{"timestamp":"2019-11-13T11:10:50.698Z","pid":1,"session_id":"5dcbe4ba.1","error_severity":"LOG","message":"listening on Unix socket \"/tmp/.s.PGSQL.5432\""}
piled b{"timestamp":"2019-11-13T11:10:50.717Z","pid":7,"session_id":"5dcbe4ba.7","error_severity":"LOG","message":"database system was shut down at 2019-11-13 11:08:45 UTC"}
 gcc (Al{"timestamp":"2019-11-13T11:10:50.725Z","pid":1,"session_id":"5dcbe4ba.1","error_severity":"LOG","message":"database system is ready to accept connections"}
piled {"timestamp":"2019-11-13T11:10:50.743Z","pid":13,"session_id":"5dcbe4ba.d","vxid":"2/0","error_severity":"LOG","message":"pg_cron scheduler started"}
32\""}
```

**Comment by exe-dealer** [comment]
I found that this bug is because of `PIPE_HEADER_SIZE +` in write_console function
https://github.com/michaelpq/pg_plugins/blob/master/jsonlog/jsonlog.c#L153

@michaelpq what the purpose is `PIPE_HEADER_SIZE +` in write_console?

**Comment by cedlerouge** [comment]
I have the same problem when I set log_collector=off. In few words this is because many process write logs at the same time. The first line of logs are in json and after 2 or 3 lines of logs, the weird characters appear.
If you enable log_collector, the jsonlog works very well but the logs is wrtten in a log file and not in stderr. 

ref: https://www.postgresql-archive.org/Proposal-Adding-json-logging-tp6017059p6017801.html

**Comment by michaelpq** [comment]
Fixed as of f6e6686.
